lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -126,9 +113,9 @@ protected function getFilename($id)
     {
         return $this->directory
             . DIRECTORY_SEPARATOR
-            . implode(str_split(hash('sha256', $id), 2), DIRECTORY_SEPARATOR)
+            . substr(hash('sha256', $id), 0, 2)
```

Comment:
Todays filesystems do not even have this limitation of files per directory. So IMO this balancing should be opt-in. This way we could save a `hash` call per cache item and another inode that is wasted on most systems. Remember the cache should be fast, so there should not be unneeded workload for every item you save/fetch/delete etc.